### PR TITLE
Add gen_js_api 1.0.4 (for OCaml 4.05)

### PR DESCRIPTION
**Pull request by alainfrisch**
Also mark the previous version as incompatible with 4.05.

### CODE REVIEW

File: packages/gen_js_api/gen_js_api.1.0.3/opam [hunk 1]
```diff
@@ -15,4 +15,4 @@ depends: [
   "ocamlfind" {build}
   "js_of_ocaml"
 ]
-available: [ocaml-version >= "4.03.0"]
+available: [ocaml-version >= "4.03.0" & ocaml-version <= "4.05"]
```
Comment (avsm) [inline packages/gen_js_api/gen_js_api.1.0.3/opam]: There are earlier versions in the repo without a similar upper bound -- do they also need a constraint added?

Comment (alainfrisch) [inline packages/gen_js_api/gen_js_api.1.0.3/opam]: Probably, but I did not take the time to find the exact constraint.

Comment (avsm) [inline packages/gen_js_api/gen_js_api.1.0.3/opam]: it's probably worth just pushing a similar <4.05 constraint to them as well

Comment (alainfrisch) [inline packages/gen_js_api/gen_js_api.1.0.3/opam]: Cf #9855.  Actually, the proper constraint was < 4.05, not <= 4.05.

**Pull request merged**